File: java/Exercise034.java
public class Exercise034 {

    public static void main(String[] args) {
        int sum = 0;
        for (int i = 10; i <= 10_000_000; i++)
        {
            if (i == sumDigitFactorial(i))
            {
                sum += i;
            }
        }
        System.out.println(sum);
    }

    public static int sumDigitFactorial(int n)
    {
        int sum = 0, digit;
        for (char c : Integer.toString(n).toCharArray())
        {
            digit = c - '0';
            sum += factorial(digit);
        }
        return sum;
    }

    public static int factorial(int digit)
    {
        if (digit < 0 || digit >= 10) throw new IllegalArgumentException("Digit is not a single digit");
        if (digit == 0) return 1;
        return digit * factorial(digit - 1);
    }
}

File: python/exercise034.py
def factorial(n):
    if n == 0:
        return 1
    return n * factorial(n - 1)


def sum_digits_fac(n):
    sum = 0
    for c in str(n):
        sum += factorial(int(c))
    return sum


sum = 0
for i in range(10, 1_000_000):
    if i == sum_digits_fac(i):
        sum += i
print(sum)